File: offer/cpp/offer_0005.cpp
#include <string>
using namespace std;

class Solution
{
public:
    string replaceSpace(string s)
    {
        int oldLength = s.length();
        int spaceCount = 0;
        for (int i = 0; i < oldLength; i++)
        {
            if (s[i] == ' ')
            {
                spaceCount++;
            }
        }
        int newLength = oldLength + 2 * spaceCount;
        s.resize(newLength);
        int i = oldLength - 1;
        int j = newLength - 1;
        while (i >= 0 && i != j)
        {
            if(s[i] == ' ') {
                s[j] = '0';
                s[j-1] = '2';
                s[j-2] = '%';
                i--;
                j -= 3;
            } else {
                s[j] = s[i];
                i--;
                j--;
            }
        }
        return s;
    }
};

File: offer/java/offer_0005.java
class Solution {
    public String replaceSpace(String s) {
        int oldLength = s.length();
        int spaceCount = 0;
        for (int i = 0; i < oldLength; i++)
        {
            if (s.charAt(i) == ' ')
            {
                spaceCount++;
            }
        }
        int newLength = oldLength + 2 * spaceCount;
        char[] array = new char[newLength];
        int i = oldLength - 1;
        int j = newLength - 1;
        while (i >= 0)
        {
            if(s.charAt(i) == ' ') {
                array[j] = '0';
                array[j-1] = '2';
                array[j-2] = '%';
                i--;
                j -= 3;
            } else {
                array[j] = s.charAt(i);
                i--;
                j--;
            }
        }
        return new String(array);
    }
}